aip/cloud/2510.md

```diff
@@ -67,6 +67,46 @@ considered PII and user data, but project numbers are not.
 Therefore, when an internal service calls an external Google APIs, it
 **should** use project numbers for making API requests.
 
+### Resource References
+
+Project IDs also appear in [resource names][], which are used
```

Comment:
Would something like this be simpler?

Project IDs may be represented as:

* project-id: "foo"
* project-number: 12345

Services that return resources with project IDs in their names **should** preserve the Project ID format in the returned response. 

Example:
a client request `projects/foo/books/game-of-tones` would result in:

```proto
book {
  name: "project/foo/books/game-of-tones" // project id is preserved
  shelf: "project/123456/shelves/top-row" // project number can be used
}
```
Note: The project ID preservation is not required for fields other than the resource name.

Reply:
The formatting is nice, but I think the intent is different.

we should be strict with respect to whether the user input (e.g. a project ID or project number based string) is unmodified or not. 

for resource references, they shouldn't be: they should be whatever the user sent.

Reply:
resolving given outdated PR.